src/intro_gui.cpp

```diff
@@ -33,13 +39,207 @@
 
 #include "safeguards.h"
 
+
+/**
+ * A viewport command for the main menu background (intro game).
+ */
+struct IntroGameViewportCommand {
+	/** Horizontal alignment value. */
+	enum AlignmentH : byte {
+		LEFT,
+		CENTRE,
+		RIGHT,
+	};
+	/** Vertical alignment value. */
+	enum AlignmentV : byte {
+		TOP,
+		MIDDLE,
+		BOTTOM,
+	};
+
+	int command_index;   ///< Sequence number of the command (performance order).
+	Point position;      ///< Calculated world coordinate to position viewport top-left at.
+	VehicleID vehicle;   ///< Vehicle to follow, or INVALID_VEHICLE if not following a vehicle.
+	uint delay;          ///< Delay until next command.
+	int zoom_adjust;     ///< Adjustment to zoom level from base zoom level.
+	bool pan_to_next;    ///< If true, do a smooth pan from this position to the next.
+	AlignmentH align_h;  ///< Horizontal alignment.
+	AlignmentV align_v;  ///< Vertical alignment.
+
+	IntroGameViewportCommand() : command_index(0), vehicle(INVALID_VEHICLE), delay(0), zoom_adjust(0), pan_to_next(false), align_h(CENTRE), align_v(MIDDLE)
```

Comment:
for constants, I think we were switching to `bool pan_to_next = false` instead of ctor? I can be wrong :P

Reply:
I haven't read up on "implicit" constructors for constant initialisation. If that's possible I'll prefer that.